### PR TITLE
Tooling

### PR DESCRIPTION
**Pull request by rousseab**
This PR simplifies the main running script by creating/configuring the pl logging, console logging and callbacks somewhere else. 

The module `crystal_diffusion/data/diffusion/data_preprocess.py` is refactored so that it is easier to create fake data for tests without having to read yaml files, which can be slow). 

The module `tests/test_train_diffusion.py` smoke tests that the diffusion training script runs and that an interrupted training can be restarted. It's not clear if the loggers cleanly pick up where they left off... we may have to do more work to make everything seamless. I'll leave that to another day!

### CODE REVIEW

File: crystal_diffusion/callbacks/standard_callbacks.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        dirpath=last_model_path,
+        filename="last_model-{epoch:03d}-{step:06d}",
+        verbose=verbose,
+        every_n_epochs=1,
+    )
+    return dict(best_checkpoint=best_checkpoint_callback, last_checkpoint=last_checkpoint_callback)
+
+
+class CustomProgressBar(RichProgressBar):
+    """A custom progress bar based on Rich that doesn't log the v_num stuff."""
```
Comment (sblackburn86) [inline crystal_diffusion/callbacks/standard_callbacks.py:52]: cool stuff, I didn't know that was a thing in lightning.

File: crystal_diffusion/main_utils.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        return early_stopping_params['metric'], early_stopping_params['mode']
+
+    else:
+        return None, None
+
+
+def get_crash_metric_result(hyper_params: Dict[AnyStr, Any]) -> MetricResult:
+    """Get crash metric result.
+
+    Produce a MetricResult object that is appropriate for reporting when there is a run time error during training..
```
Comment (sblackburn86) [inline crystal_diffusion/main_utils.py:41]: double period at EOS

File: crystal_diffusion/main_utils.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        # The metric must be minimized; this is already aligned with what Orion will optimize.
+        optimization_objective_name = metric_name
+        optimization_sign = 1
+    else:
+        raise ValueError("The mode for this early_stopping_metric is unknown")
+    return optimization_objective_name, optimization_sign
+
+
+def report_to_orion_if_on(metric_result: MetricResult, run_time_error: Union[None, RuntimeError]):
+    """Report to orion if on.
```
Comment (sblackburn86) [inline crystal_diffusion/main_utils.py:85]: capitalize Orion to be consistent

**Review by sblackburn86 — APPROVED** [review]

File: crystal_diffusion/main_utils.py (hunk 2 above)
Comment (rousseab) [reply crystal_diffusion/main_utils.py]: fixed.

File: crystal_diffusion/main_utils.py (hunk 3 above)
Comment (rousseab) [reply crystal_diffusion/main_utils.py]: fixed.

**Pull request merged**